sw/device/silicon_creator/lib/boot_log.h

```diff
@@ -11,24 +11,45 @@
 #include "sw/device/silicon_creator/lib/chip_info.h"
 #include "sw/device/silicon_creator/lib/drivers/hmac.h"
 #include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/nonce.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * The boot_log encodes information about how the chip booted.
+ */
 typedef struct boot_log {
+  /** Digest to indicate validity of the boot_log. */
   hmac_digest_t digest;
+  /** Identifier (`BOLG`). */
   uint32_t identifier;
+  /** Chip version (from the ROM). */
   chip_info_scm_revision_t chip_version;
+  /** Which ROM_EXT slot booted. */
   uint32_t rom_ext_slot;
```

Comment:
Do you think we should report the size of ROM_EXT? If we remove the 64KB restriction, I think it would be nice if we can potentially give some space back to the owner. Although at the same time, I don't want to make ROM_EXT sustainability harder. 

We can maybe consider downsizing to 32KB if the cryptolib is distributed as a library that the owner can integrate into its binary.

Reply:
Added.